### PR TITLE
ext_proc: Pass stream_info to gRPC streams

### PR DESCRIPTION
**Pull request by gbrail**
Commit Message: The ext_proc filter was not passing the stream_info structure down to the gRPC stream that it creates. As a result, certain metadata keys could cause the filter to crash when trying to open the stream.
Additional Description:
Risk Level: Low. Makes the filter more resilient.
Testing: New code in integration test failed before this fix was applied and now passes.

### CODE REVIEW

File: test/extensions/filters/http/ext_proc/streaming_integration_test.cc [hunk 1]
```diff
@@ -69,6 +69,12 @@ class StreamingIntegrationTest : public HttpIntegrationTest,
       const auto addr = Network::Test::getCanonicalLoopbackAddress(ipVersion());
       const auto addr_port = Network::Utility::getAddressWithPort(*addr, test_processor_.port());
       setGrpcService(*proto_config_.mutable_grpc_service(), "ext_proc_server", addr_port);
+      // Insert some extra metadata. This ensures that we are actually passing the
+      // "stream info" from the original HTTP request all the way down to the
+      // ext_proc stream.
+      auto* metadata = proto_config_.mutable_grpc_service()->mutable_initial_metadata()->Add();
```
Comment (tonya11en) [inline test/extensions/filters/http/ext_proc/streaming_integration_test.cc:75]: How does the test verify that the metadata gets passed down?

Comment (gbrail) [reply test/extensions/filters/http/ext_proc/streaming_integration_test.cc:75]: I was focused on ensuring that it doesn't crash, and not focusing on whether the underlying Envoy gRPC library sends the metadata to the server. But I added some test code to verify that as well. Thanks!

Comment (snowp) [inline test/extensions/filters/http/ext_proc/streaming_integration_test.cc:75]: Just so I understand, how is this related to stream info? I would have expected this to be used when the grpc stream is created by reading the config, not sure how this would be plumbed though via StreamInfo?

**Review by snowp — COMMENTED** [review]
Just one question on how this all fits together, but I think this looks good

**Comment by gbrail** [comment]
I don't know the ins and outs of metadata and our gRPC client, but I think
that "%REQ" syntax means, "pull information from the original request and
add it to the stream." The grpc client seems to do that using the
stream_info, and since I wasn't setting that when opening the gRPC stream
to the external processor, it was crashing while trying to get that
information.

**Comment by snowp** [comment]
Got it, I didn't realize that the important bit was the REQ query, thanks!

Seems like this will need a main merge

**Comment by gbrail** [comment]
No problem -- I had two in flight at the same time. Let me do that and
you'll see it pop up.

**Review by snowp — APPROVED** [review]
Thanks!
